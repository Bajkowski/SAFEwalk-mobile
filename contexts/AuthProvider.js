import React, { createContext, useReducer, useMemo } from "react";
import { AsyncStorage } from "react-native";

export const AuthContext = createContext();

export function AuthProvider({ children }) {
  const [state, dispatch] = useReducer(authReducer, {
    // initial states
    isLoading: true,
    isSignout: false,
    userToken: null,
    userType: "user",
    userEmail: null
  });

  // Handle Login
  async function login(userType, { email, password }) {
    try {
      // Put code below for this: make login GET request to server

      // For now, while server is not set, use email as value for user token
      await AsyncStorage.setItem("userToken", JSON.stringify(email));
      // Store the user type i.e. which type of user is logged in, user or SAFEwalker.
      await AsyncStorage.setItem("userType", userType);
      // Store the user email
      await AsyncStorage.setItem("userEmail", JSON.stringify(email));

      dispatch({
        type: "LOG_IN",
        token: JSON.stringify(email),
        userType: userType,
        userEmail: email
      });
    } catch (error) {
      throw new Error("Error in login(): " + error);
    }
  }

  // Handle signout
  async function signout() {
    try {
      await AsyncStorage.removeItem("userToken");
      await AsyncStorage.removeItem("userType");
      await AsyncStorage.removeItem("userEmail");

      dispatch({ type: "SIGN_OUT" });
    } catch (error) {
      throw new Error("Error in signout(): " + error);
    }
  }

  // Handle register
  async function register({
    email,
    password,
    firstName,
    lastName,
    phoneNumber
  }) {
    try {
      // Put code below for this: make register POST request to server

      // For now, while server is not set, use email as value for user token
      await AsyncStorage.setItem("userToken", JSON.stringify(email));
      // Store the user type i.e. which type of user is logged in, user or SAFEwalker.
      await AsyncStorage.setItem("userType", userType);
      // Store the user email
      await AsyncStorage.setItem("userEmail", JSON.stringify(email));

      dispatch({
        type: "LOG_IN",
        token: JSON.stringify(email),
        userType: "user",
        userEmail: email
      });
    } catch (error) {
      throw new Error("Error in register(): " + error);
    }
  }

  const authContextValue = useMemo(() => {
    return {
      isLoading: state.isLoading,
      isSignout: state.isSignout,
      userToken: state.userToken,
      userType: state.userType,
      userEmail: state.userEmail,
      dispatch,
      login,
      signout,
      register
    };
  }, [state]);

  return (
    <AuthContext.Provider value={authContextValue}>
      {children}
    </AuthContext.Provider>
  );
}

function authReducer(prevState, action) {
  switch (action.type) {
    case "RESTORE_TOKEN":
      return {
        ...prevState,
        isLoading: false,
        userToken: action.token,
        userType: action.userType,
        userEmail: action.userEmail
      };
    case "LOG_IN":
      return {
        ...prevState,
        isSignout: false,
        userToken: action.token,
        userType: action.userType,
        userEmail: action.userEmail
      };
    case "SIGN_OUT":
      return {
        ...prevState,
        isSignout: true,
        userToken: null,
        userType: null,
        userEmail: null
      };
    default: {
      throw new Error(`Unhandled auth action type: ${action.type}`);
    }
  }
}
