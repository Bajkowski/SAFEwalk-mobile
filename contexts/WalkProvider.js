import React, { createContext, useReducer, useMemo } from "react";
import { AsyncStorage } from "react-native";

export const WalkContext = createContext();

export function WalkProvider({ children }) {
  const [state, dispatch] = useReducer(walkReducer, {
    // initial states
    isWalkLoading: true,
    isWalkActive: "false", // has to be a string due to AsyncStorage constraints (cannot store boolean)

    // initial walk state
    walkId: null,

    // initial user state
    userEmail: null,
    walkerEmail: null,

    // initial SAFEwalker state
    userSocketId: null,
    walkerSocketId: null,
  });

  // Set up walk ID
  async function setWalkAsActive() {
    try {
      // Store walk ID
      await AsyncStorage.setItem("isWalkActive", "true");

      dispatch({
        type: "SET_WALK_ACTIVE",
        isWalkActive: "true",
      });
    } catch (error) {
      console.error("Error in setIsWalkActive(): " + error);
    }
  }

  // Set up walk ID
  async function setWalkId(walkId) {
    try {
      // Store walk ID
      await AsyncStorage.setItem("walkId", walkId);

      dispatch({
        type: "SET_WALK_ID",
        walkId: walkId,
      });
    } catch (error) {
      console.error("Error in setWalkId(): " + error);
    }
  }

  // Remove walk ID
  async function removeWalkId() {
    try {
      // remove walk ID
      await AsyncStorage.removeItem("WalkId");

      dispatch({ type: "REMOVE_WALK_ID" });
    } catch (error) {
      console.error("Error in removeWalkId(): " + error);
    }
  }

  // Store User information (walkId, email and socket ID)
  async function setUserInfo(walkId, userEmail, userSocketId) {
    try {
      await AsyncStorage.setItem("walkId", walkId);
      await AsyncStorage.setItem("userEmail", userEmail);
      await AsyncStorage.setItem("userSocketId", userSocketId);

      dispatch({
        type: "SET_USER_INFO",
        walkId: walkId,
        userEmail: userEmail,
        userSocketId: userSocketId,
      });
    } catch (error) {
      console.error("Error in seUserInfo(): " + error);
    }
  }

  // Store Safewalker information (email and socket ID)
  async function setWalkerInfo(walkerEmail, walkerSocketId) {
    try {
      await AsyncStorage.setItem("walkerEmail", walkerEmail);
      await AsyncStorage.setItem("walkerSocketId", walkerSocketId);

      dispatch({
        type: "SET_WALKER_INFO",
        walkerEmail: walkerEmail,
        walkerSocketId: walkerSocketId,
      });
    } catch (error) {
      console.error("Error in setWalkerInfo(): " + error);
    }
  }

  // remove all current walk-related information
  async function resetWalkContextState() {
    try {
      await AsyncStorage.removeItem("walkId");
      await AsyncStorage.removeItem("userEmail");
      await AsyncStorage.removeItem("walkerEmail");
      await AsyncStorage.removeItem("userSocketId");
      await AsyncStorage.removeItem("walkerSocketId");

      dispatch({ type: "RESET" });
    } catch (error) {
      console.error("Error in resetContextState(): " + error);
    }
  }

  const walkContextValue = useMemo(() => {
    return {
      // states
      isWalkLoading: state.isWalkLoading,
      isWalkActive: state.isWalkActive,
      walkId: state.walkId,
      userEmail: state.userEmail,
      walkerEmail: state.walkerEmail,
      userSocketId: state.userSocketId,
      walkerSocketId: state.walkerSocketId,

      // functions
      setWalkAsActive,
      setWalkId,
      removeWalkId,
      setUserInfo,
      setWalkerInfo,
      resetWalkContextState,
      dispatch,
    };
  }, [state]);

  return (
    <WalkContext.Provider value={walkContextValue}>
      {children}
    </WalkContext.Provider>
  );
}

function walkReducer(prevState, action) {
  switch (action.type) {
    case "RESTORE_STATE":
      return {
        ...prevState,
        isWalkLoading: false,
        isWalkActive: action.isWalkActive,
        walkId: action.walkId,
        userEmail: action.userEmail,
        walkerEmail: action.walkerEmail,
        userSocketId: action.userSocketId,
        walkerSocketId: action.walkerSocketId,
      };
    case "SET_WALK_ACTIVE":
      return {
        ...prevState,
        isWalkActive: "true",
      };
    case "SET_WALK_ID":
      return {
        ...prevState,
        walkId: action.walkId,
      };
    case "REMOVE_WALK_ID":
      return {
        ...prevState,
        walkId: null,
      };
    case "SET_USER_INFO":
      return {
        ...prevState,
        walkId: action.walkId,
        userEmail: action.userEmail,
        userSocketId: action.userSocketId,
      };
    case "SET_WALKER_INFO":
      return {
        ...prevState,
        walkerEmail: action.walkerEmail,
        walkerSocketId: action.walkerSocketId,
      };
    case "RESET":
      return {
        ...prevState,
        isWalkLoading: false,
        isWalkActive: "false",
        walkId: null,
        userEmail: null,
        userSocketId: null,
        walkerEmail: null,
        walkerSocketId: null,
      };

    default: {
      console.error(`Unhandled walk action type: ${action.type}`);
    }
  }
}
