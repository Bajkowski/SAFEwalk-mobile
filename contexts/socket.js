const io = require("socket.io-client");
module.exports = io("http://127.0.0.1:3000");
// module.exports = io("http://365b17d8.ngrok.io");
