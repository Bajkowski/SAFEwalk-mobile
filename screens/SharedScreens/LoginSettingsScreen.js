import React, { useEffect, useState, useContext } from "react";
import {
  StyleSheet,
  Text,
  View,
  KeyboardAvoidingView,
  TouchableWithoutFeedback,
  Keyboard,
  Alert,
} from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";
import { useForm } from "react-hook-form";
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from "react-native-responsive-screen";

// Constants
import url from "./../../constants/api";
import colors from "./../../constants/colors";
import TextInput from "./../../components/TextInput";
import Button from "./../../components/Button";
import style from "./../../constants/style";

// Contexts
import { AuthContext } from "./../../contexts/AuthProvider";
import { BorderlessButton } from "react-native-gesture-handler";

export default function LoginSettingsScreen({ navigation }) {
  const { userToken, email, userType } = useContext(AuthContext);

  // forms input handling
  const { register, setValue, handleSubmit, errors, watch } = useForm();

  // password input upon change
  useEffect(() => {
    register("currentPassword");
    register("password");
    register("confirmPassword");
  }, [register]);

  // upon clicking update password button
  const saveProfileInfo = async (data) => {
    // first check old password
    let isUser = true;
    if (userType === "safewalker") {
      isUser = false;
    }
    let endpoint = "/api/Login/";
    let oldPass = 0;

    // checking old password with database
    const response1 = await fetch(url + endpoint + email + "/PasswordVerify", {
      method: "GET",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
        password: data.currentPassword,
        isUser: isUser,
      },
    }).then((response1) => {
      if (!(response1.status === 200)) {
        oldPass = 0;
        console.log("captured " + response1.status + "! Try again.");
      } else {
        // only updates to 1 if password is correct
        oldPass = 1;
      }
    });

    endpoint = "/api/Users/";
    if (userType == "safewalker") {
      endpoint = "/api/Safewalkers/";
    }

    if (oldPass == 1) {
      // send new info to the database
      const response = await fetch(url + endpoint + email, {
        method: "PUT",
        headers: {
          Accept: "application/json",
          "Content-Type": "application/json",
          token: userToken,
        },
        body: JSON.stringify({
          Password: data.confirmPassword,
        }),
      })
        .then((response) => {
          if (!(response.status === 200)) {
            console.log("captured " + response.status + "! Try again.");
          } else {
            console.log("updated password" + data.confirmPassword);
            Alert.alert("Password Successfully Updated", "", [{ text: "OK" }], {
              cancelable: false,
            });
          }
        })
        .catch((error) => {
          console.log(error.message);
          console.log("Error in updating password. Please try again.");
        });
    } else {
      Alert.alert(
        "Incorrect Password",
        "Please Confirm Password",
        [{ text: "OK" }],
        { cancelable: false }
      );
    }
  };

  // upon pressing the update password button
  const onSubmit = (data) => {
    const response = saveProfileInfo(data);
  };

  return (
    <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
      <View style={styles.container}>
        <SafeAreaView style={styles.container2}>
          {/* Top View */}
          <View style={styles.containerTop}>
            <Text style={styles.textTitle}> Update Password:</Text>
          </View>

          {/* Inner View */}
          <KeyboardAvoidingView style={styles.innerContainer}>
            {errors.currentPassword && (
              <Text style={styles.textError}>
                Current password is required.
              </Text>
            )}
            <TextInput
              label="Current Password"
              placeholder="Current Password"
              ref={register({ name: "currentPassword" }, { required: true })}
              onChangeText={(text) => setValue("currentPassword", text, true)}
              secureTextEntry
              style={styles.textInput}
            />

            {errors.password && (
              <Text style={styles.textError}>Password is required.</Text>
            )}
            <TextInput
              label="New Password"
              placeholder="New Password"
              ref={register({ name: "password" }, { required: true })}
              onChangeText={(text) => setValue("password", text, true)}
              secureTextEntry
              style={styles.textInput}
            />

            {errors.confirmPassword && (
              <Text style={styles.textError}>The passwords do not match.</Text>
            )}
            <TextInput
              label="Confirm password"
              ref={register(
                { name: "confirmPassword" },
                {
                  required: true,
                  validate: (value) =>
                    value === watch("password") ||
                    "The passwords do not match.",
                }
              )}
              onChangeText={(text) => setValue("confirmPassword", text, true)}
              secureTextEntry
              style={styles.textInput}
            />

            {/* Bottom */}
            <View style={styles.containerButton}>
              <Button
                title="Confirm Password Change"
                onPress={handleSubmit(onSubmit)}
              />
            </View>
          </KeyboardAvoidingView>
        </SafeAreaView>
      </View>
    </TouchableWithoutFeedback>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.white,
  },
  container2: {
    flex: 1,
    marginHorizontal: style.marginContainerHorizontal,
    marginVertical: hp("6%"),
  },
  containerTop: {
    height: hp("10%"),
    justifyContent: "space-around",
    fontSize: wp("4%"),
  },
  containerButton: {
    height: hp("10%"),
    justifyContent: "space-around",
  },
  innerContainer: {
    justifyContent: "space-around",
    height: hp("40%"),
  },
  textError: {
    color: colors.red,
    alignSelf: "center",
    fontSize: wp("4%"),
  },
  textTitle: {
    color: colors.orange,
    fontSize: style.fontSize,
    fontWeight: "bold",
  },
  textInput: {
    marginBottom: 20,
  },
});
