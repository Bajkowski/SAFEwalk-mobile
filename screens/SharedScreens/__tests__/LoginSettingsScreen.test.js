import React from "react";
import { create, shallow } from "react-test-renderer";
import renderer from "react-test-renderer";
import { View, Button } from "react-native";
import {
  fireEvent,
  NativeTestEvent,
  act,
  wait,
  waitForElementToBeRemoved,
  render,
} from "@testing-library/react-native";

import { NavigationContainer } from "@react-navigation/native";

import LoginSettingsScreen from "../LoginSettingsScreen";
import { AuthProvider } from "../../../contexts/AuthProvider";
import { WalkProvider } from "../../../contexts/WalkProvider";

const AbortController = require("abort-controller");

jest.mock("abort-controller", () => {
  return { AbortController: () => jest.fn() };
});

jest.mock("react-native-safe-area-context", () => ({
    SafeAreaView: ({ children }) => <>{children}</>,
  }));

  export const createTestProps = (props) => ({
    navigation: {
      navigate: jest.fn(),
      goBack: jest.fn(),
      replace: jest.fn(),
    },
    ...props,
  });
  
  let props;
  let component;
  let screen;

describe("LoginSettingsScreen ", () => {
    beforeEach(() => {
        props = createTestProps();
    
        let route = {
          params: {
            email: "a@wisc.edu",
            password: "asdfg",
          },
        };
    
        component = (
          <AuthProvider>
            <LoginSettingsScreen route={route} navigation={props.navigation} />
          </AuthProvider>
        );
    
        screen = render(component);
      });


  it("renders correctly", () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("has a current password TextInput", () => {
    expect(screen.getByLabelText("Current Password")).toBeTruthy();
  });
 
  it("has a new password TextInput", () => {
    expect(screen.getByLabelText("New Password")).toBeTruthy();
  });

  it("has a confirm password TextInput", () => {
    expect(screen.getByLabelText("Confirm Password")).toBeTruthy();
  });

  it("has a save password button", () => {
    expect(screen.getByLabelText("Save Button")).toBeTruthy();
  });

  it("shows error if current password is empty", async () => {
    let saveButton = screen.getByLabelText("Save Button");
    act(() => {
      fireEvent.press(saveButton);
    });

    // wait for appearance
    await wait(() => {
      expect(screen.getByLabelText("currentRequired")).toBeTruthy();
      expect(screen.getByLabelText("newRequired")).toBeTruthy();
      expect(screen.getByLabelText("confirmRequired")).toBeTruthy();
    });
  });

  it("shows error if current password is incorrect", () => {
    act(() => {
      fireEvent.changeText(screen.getByLabelText("Current Password"), "123");
    });

    act(() => {
      fireEvent.changeText(screen.getByLabelText("New Password"), "asdf");
    });

    act(() => {
        fireEvent.changeText(screen.getByLabelText("Confirm Password"), "asdf");
      });

      let saveButton = screen.getByLabelText("Save Button");
      act(() => {
        fireEvent.press(saveButton);
      });
  });

  it("shows error if new password does not match confirmation", () => {
    act(() => {
        fireEvent.changeText(screen.getByLabelText("New Password"), "asdfd");
      });

    act(() => {
        fireEvent.changeText(screen.getByLabelText("Confirm Password"), "asdf");
      });
  });


  it("saves the password upon success", async () => {
    let response = {
        status: 200,
    };

    global.fetch = jest.fn(() => Promise.resolve(response));

    // Change inputs
    act(() => {
        fireEvent.changeText(screen.getByLabelText("Current Password"), "asdfg");
    });
    act(() => {
        fireEvent.changeText(screen.getByLabelText("New Password"), "pass");
    });
    act(() => {
        fireEvent.changeText(screen.getByLabelText("Confirm Password"), "pass");
    });

    // press save button
    let saveButton = screen.getByLabelText("Save Button");
    act(() => {
        fireEvent.press(saveButton);
    });
});

  it("Alerts the user upon failure to update password", async () => {
    let response = {
        status: 409,
    };

    global.fetch = jest.fn(() => Promise.resolve(response));

    // Change inputs
    act(() => {
        fireEvent.changeText(screen.getByLabelText("Current Password"), "password");
    });
    act(() => {
        fireEvent.changeText(screen.getByLabelText("New Password"), "pass");
    });
    act(() => {
        fireEvent.changeText(screen.getByLabelText("Confirm Password"), "pass");
    });

    // press save button
    let saveButton = screen.getByLabelText("Save Button");
    act(() => {
        fireEvent.press(saveButton);
    });
});

it("shows error if there's other uncaptured server errors", async () => {
    global.fetch = jest.fn(() => Promise.resolve(response));

    // Change inputs
    act(() => {
        fireEvent.changeText(screen.getByLabelText("Current Password"), "asdfg");
    });
    act(() => {
        fireEvent.changeText(screen.getByLabelText("New Password"), "pass");
    });
    act(() => {
        fireEvent.changeText(screen.getByLabelText("Confirm Password"), "pass");
    });

    // press save button
    let saveButton = screen.getByLabelText("Save Button");
    act(() => {
        fireEvent.press(saveButton);
    });
});

it("fails if email and password is null", async () => {
    let route = {
      params: {
        email: null,
        password: null,
      },
    };

    component = (
      <AuthProvider>
        <LoginSettingsScreen route={route} navigation={props.navigation} />
      </AuthProvider>
    );

    screen = render(component);
  });

});
