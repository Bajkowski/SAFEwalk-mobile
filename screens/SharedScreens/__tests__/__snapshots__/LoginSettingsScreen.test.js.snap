// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginSettingsScreen  renders correctly 1`] = `
<TouchableWithoutFeedback
  accessible={false}
  onPress={[Function]}
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "fontSize": 30,
          "height": 133.5,
          "justifyContent": "space-around",
          "marginVertical": 40,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fd6e11",
            "fontSize": 24,
            "fontWeight": "bold",
            "marginBottom": 40,
            "textAlign": "center",
          }
        }
      >
         Update Password
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "height": 533.5,
          "justifyContent": "space-around",
        }
      }
    >
      <View
        style={
          Object {
            "marginBottom": 20,
          }
        }
      >
        <View>
          <View
            pointerEvents="none"
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                },
                Object {
                  "backgroundColor": "#f6f6f6",
                  "borderColor": "rgba(0, 0, 0, 0.54)",
                  "borderRadius": 4,
                  "borderWidth": 1,
                },
              ]
            }
          />
          <View
            style={
              Object {
                "minHeight": 80,
                "paddingBottom": 0,
                "paddingTop": 8,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "#f6f6f6",
                    "bottom": 0,
                    "height": 2,
                    "left": 10,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                    "width": 8,
                  }
                }
              />
              <Text
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "#f6f6f6",
                    "color": "transparent",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 18,
                    "opacity": 0,
                    "paddingHorizontal": 0,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 49,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                      Object {
                        "scaleY": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Current Password
              </Text>
              <Text
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#fd6e11",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 48,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Current Password
              </Text>
              <Text
                numberOfLines={1}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 1,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 48,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Current Password
              </Text>
            </View>
            <TextInput
              accessibilityLabel="Current Password"
              allowFontScaling={true}
              editable={true}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder=""
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              rejectResponderTermination={true}
              secureTextEntry={true}
              selectionColor="#fd6e11"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                    "paddingHorizontal": 14,
                    "textAlign": "left",
                    "zIndex": 1,
                  },
                  Object {
                    "height": 80,
                  },
                  Object {
                    "paddingBottom": 32,
                    "paddingTop": 32,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "textAlignVertical": "center",
                  },
                ]
              }
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 20,
          }
        }
      >
        <View>
          <View
            pointerEvents="none"
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                },
                Object {
                  "backgroundColor": "#f6f6f6",
                  "borderColor": "rgba(0, 0, 0, 0.54)",
                  "borderRadius": 4,
                  "borderWidth": 1,
                },
              ]
            }
          />
          <View
            style={
              Object {
                "minHeight": 80,
                "paddingBottom": 0,
                "paddingTop": 8,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "#f6f6f6",
                    "bottom": 0,
                    "height": 2,
                    "left": 10,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                    "width": 8,
                  }
                }
              />
              <Text
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "#f6f6f6",
                    "color": "transparent",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 18,
                    "opacity": 0,
                    "paddingHorizontal": 0,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 49,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                      Object {
                        "scaleY": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                New Password
              </Text>
              <Text
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#fd6e11",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 48,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                New Password
              </Text>
              <Text
                numberOfLines={1}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 1,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 48,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                New Password
              </Text>
            </View>
            <TextInput
              accessibilityLabel="New Password"
              allowFontScaling={true}
              editable={true}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder=""
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              rejectResponderTermination={true}
              secureTextEntry={true}
              selectionColor="#fd6e11"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                    "paddingHorizontal": 14,
                    "textAlign": "left",
                    "zIndex": 1,
                  },
                  Object {
                    "height": 80,
                  },
                  Object {
                    "paddingBottom": 32,
                    "paddingTop": 32,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "textAlignVertical": "center",
                  },
                ]
              }
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 20,
          }
        }
      >
        <View>
          <View
            pointerEvents="none"
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                },
                Object {
                  "backgroundColor": "#f6f6f6",
                  "borderColor": "rgba(0, 0, 0, 0.54)",
                  "borderRadius": 4,
                  "borderWidth": 1,
                },
              ]
            }
          />
          <View
            style={
              Object {
                "minHeight": 80,
                "paddingBottom": 0,
                "paddingTop": 8,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "#f6f6f6",
                    "bottom": 0,
                    "height": 2,
                    "left": 10,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                    "width": 8,
                  }
                }
              />
              <Text
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "#f6f6f6",
                    "color": "transparent",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 18,
                    "opacity": 0,
                    "paddingHorizontal": 0,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 49,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                      Object {
                        "scaleY": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Confirm password
              </Text>
              <Text
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#fd6e11",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 48,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Confirm password
              </Text>
              <Text
                numberOfLines={1}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 1,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 48,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Confirm password
              </Text>
            </View>
            <TextInput
              accessibilityLabel="Confirm Password"
              allowFontScaling={true}
              editable={true}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder=""
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              rejectResponderTermination={true}
              secureTextEntry={true}
              selectionColor="#fd6e11"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                    "paddingHorizontal": 14,
                    "textAlign": "left",
                    "zIndex": 1,
                  },
                  Object {
                    "height": 80,
                  },
                  Object {
                    "paddingBottom": 32,
                    "paddingTop": 32,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "textAlignVertical": "center",
                  },
                ]
              }
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 133.5,
            "justifyContent": "flex-end",
            "marginTop": 53.5,
          }
        }
      >
        <View
          style={
            Object {
              "borderRadius": 3,
            }
          }
        >
          <TouchableOpacity
            accessibilityLabel="Save Button"
            activeOpacity={0.3}
            disabled={false}
            onPress={[Function]}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fd6e11",
                  "borderColor": "#2089dc",
                  "borderRadius": 5,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 80,
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 26.5,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Confirm Password Change
              </Text>
            </View>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  </View>
</TouchableWithoutFeedback>
`;
